src/Form/widgets/GeoWidget/GeoWidget.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+
+import GeoMap from './components/GeoMap';
+
+import {LatLngExpression} from 'leaflet';
+import isEmpty from 'lodash/isEmpty';
+
+import {IWidget} from '../../../typings/IWidget';
+import Errors from '../../components/Errors';
+import validator from '../../Validator';
+
+interface TGeoWidgetState {
+  value?: LatLngExpression | undefined | null;
+  errors: any[];
+}
+
+export default class GeoWidget extends React.Component<
+  IWidget,
+  TGeoWidgetState
+> {
+  public static defaultProps = {
+    uiSchema: {},
+    isRequired: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {
+      value,
+      schema: {properties},
+    } = props;
+
+    if (!isEmpty(value)) {
+      this.state = {
+        errors: [],
+        value,
+      };
+    } else {
+      this.state = {
+        errors: [],
+        value: {
+          lat: properties.lat.default,
+          lng: properties.lng.default,
+        },
+      };
+    }
+  }
+
+  public get value(): LatLngExpression {
+    return this.state.value;
+  }
+
+  public get isValid(): boolean {
+    const {schema} = this.props;
+    const errors = [];
+
+    validator.validate(schema, this.value);
+
+    if (validator.errors) {
+      errors.push(...validator.errors);
+    }
+
+    this.setState({errors});
+
+    return errors.length === 0;
+  }
+
+  public render() {
+    return (
+      <React.Fragment>
+        <Errors errors={this.state.errors} />
+        <GeoMap onChange={this.handleChange} value={this.state.value} />
+      </React.Fragment>
+    );
+  }
+
+  private handleChange = (value: LatLngExpression): void => {
+    this.setState({value});
```

Comment:
Please, add some value converting because value from GeoMap is `string` and validation requires `number`.

![screenshot from 2018-10-02 12-29-40](https://user-images.githubusercontent.com/28134389/46345674-bb2afd80-c645-11e8-99c6-10734e4ce6a8.png)